packages/apple-pay-integration/src/apple-pay-payment-strategy.ts

```diff
@@ -235,4 +247,44 @@ export default class ApplePayPaymentStrategy implements PaymentStrategy {
             );
         }
     }
+
+    private async _getBraintreeApplePaySessionId(applePayPaymentMethod: PaymentMethod) {
+        if (applePayPaymentMethod.initializationData?.gateway === ApplePayGatewayType.BRAINTREE) {
```

Comment:
Please move this condition to the place where you are calling _getBraintreeApplePaySessionId method